api/v1alpha1/externaldns_webhook.go

```diff
@@ -0,0 +1,81 @@
+/*
+Copyright 2021.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"errors"
+	"fmt"
+	"regexp"
+
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/webhook"
+)
+
+// webhookLog is for logging in this package.
+var webhookLog = logf.Log.WithName("externaldns-resource")
+
+func (r *ExternalDNS) SetupWebhookWithManager(mgr ctrl.Manager) error {
+	return ctrl.NewWebhookManagedBy(mgr).
+		For(r).
+		Complete()
+}
+
+//+kubebuilder:webhook:path=/validate-externaldns-olm-openshift-io-v1alpha1-externaldns,mutating=false,failurePolicy=fail,sideEffects=None,groups=externaldns.olm.openshift.io,resources=externaldns,verbs=create;update,versions=v1alpha1,name=vexternaldns.kb.io,admissionReviewVersions={v1,v1beta1}
+
+var _ webhook.Validator = &ExternalDNS{}
+
+// ValidateCreate implements webhook.Validator so a webhook will be registered for the type
+func (r *ExternalDNS) ValidateCreate() error {
+	webhookLog.Info("validate create", "name", r.Name)
+	return r.validateFilters()
```

Comment:
Can we add more validations?
- `FQDNTemplate` has to be there if HostnameAnnotationPolicy is `Ignore`
- Credentials/configs have to be there for each provider